<a href="https://colab.research.google.com/github/ananya0504/MLproject/blob/main/Ananya_Gupta.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**---------------------DAY 6-----------------**

In [4]:
import pandas as pd
import numpy as np

In [5]:
from sklearn.datasets import load_iris #sklearn is the framework, datasets is the module and load_iris is the function
iris_dataset = load_iris()

In [6]:
print("Keys of iris_dataset are : \n{}".format(iris_dataset.keys()) )

Keys of iris_dataset are : 
dict_keys(['data', 'target', 'target_names', 'DESCR', 'feature_names', 'filename'])


In [7]:
val = iris_dataset['DESCR']
start_val= val[:200]
print(start_val + "\n...")

.. _iris_dataset:

Iris plants dataset
--------------------

**Data Set Characteristics:**

    :Number of Instances: 150 (50 in each of three classes)
    :Number of Attributes: 4 numeric, predictive
...


In [8]:
print("Target names : {}".format(iris_dataset['target_names']) ) #the species of flowers we want to predict

Target names : ['setosa' 'versicolor' 'virginica']


In [9]:
print("Feature names : {}".format(iris_dataset['feature_names']) ) #the description of each feature of a flower

Feature names : ['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)', 'petal width (cm)']


In [10]:
print("Type of data : {}".format(type(iris_dataset['data']))) #the type in which we recieve data

Type of data : <class 'numpy.ndarray'>


In [11]:
print("Shape of data : {}".format(iris_dataset['data'].shape)) #the rows represnt 150 flowers while the columns represent the 4 feature measurements
#Individual items are called samples or data points and their properties are called features
#shape is the number of samples multiplied by the number of features

Shape of data : (150, 4)


In [12]:
print("First five columns of data : \n{}".format(iris_dataset['data'][:5]))  # we observe that the all the flowers have the same petal width and the 
# first flower has the max sepal width

First five columns of data : 
[[5.1 3.5 1.4 0.2]
 [4.9 3.  1.4 0.2]
 [4.7 3.2 1.3 0.2]
 [4.6 3.1 1.5 0.2]
 [5.  3.6 1.4 0.2]]


In [13]:
print("Type of Target : {}".format(type(iris_dataset['target']))) #the type in which we recieve the target

Type of Target : <class 'numpy.ndarray'>


In [14]:
print("Shape of Target : {}".format(iris_dataset['target'].shape)) #we see that is has one entry per flower

Shape of Target : (150,)


In [15]:
print("Target : {}".format(iris_dataset['target'])) #we see that the species are encoded as digits between 0-2
#0-sentose 1-versicolor 2-virginica

Target : [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 2 2 2 2 2 2 2 2 2 2 2
 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2
 2 2]


**-------------DAY7----------------------------**

In [16]:
import sklearn # most used ML framework

X->data(2D) (The 4 attributes we have)

y->label(1D) (the 3 flower labels we have)

from f(x)=y, where you provide x as input and get y as output

In [17]:
from sklearn.model_selection import train_test_split #train_test_split is a function to split data into
                                                    # 75% training data and 25% testing data by default

X_train,X_test, y_train, y_test = train_test_split(
    iris_dataset['data'], iris_dataset['target'], random_state=0) #pick the training data randomly from training the 
                                                                  # model as we have our data arranged into an orderly fashion
                                                                  #it will be same random order for sir as well us the rest of us

In [18]:
print("X_train shape : {}".format(X_train.shape))
print("y_train shape : {}".format(y_train.shape))
print("X_test shape : {}".format(X_test.shape))
print("y_test shape : {}".format(y_test.shape))

X_train shape : (112, 4)
y_train shape : (112,)
X_test shape : (38, 4)
y_test shape : (38,)


Gives 112 data points and 4 colums for the 4 attributes


In [22]:
from sklearn.neighbors import KNeighborsClassifier #neighbors is the module and KNeighborsClassifier is a class

In [23]:
knn = KNeighborsClassifier(n_neighbors=1) # knn is the name of the object/bot that we have made

In [25]:
knn.fit(X_train, y_train) # to build the model on the training set we call the fit method of knn object which
                          #which takes as arguments the numpy array X_train and the y_train of the training data 
                          #and the corresponding training labels respectively

KNeighborsClassifier(algorithm='auto', leaf_size=30, metric='minkowski',
                     metric_params=None, n_jobs=None, n_neighbors=1, p=2,
                     weights='uniform')

In [27]:
X_new = np.array([[5,2.9,1,0.2]]) #we put the measurements of a random flower into a 2D numpy array 
                                  #as sklearn expects 2D  arrays as data

print("X_new.shape: {}".format(X_new.shape))

X_new.shape: (1, 4)


In [29]:
prediction = knn.predict(X_new) #call the predict method to make a prediction

print("Prediction: {}".format(prediction))
print("Predicted Target Name: {}".format(
    iris_dataset['target_names'][prediction]
))

Prediction: [0]
Predicted Target Name: ['setosa']


In [30]:
#Predicting for the test dataset now(X_test)

y_pred = knn.predict(X_test)
print("Test Set Predictions: \n {}".format(y_pred))

Test Set Predictions: 
 [2 1 0 2 0 2 0 1 1 1 2 1 1 1 1 0 1 1 0 0 2 1 0 0 2 0 0 1 1 0 2 1 0 2 2 1 0
 2]


In [31]:
#check if the predictions were actually correct or not
print("Test Set Score: {}".format(np.mean(y_pred == y_test))) #for all 38 values

Test Set Score: 0.9736842105263158
